src/services/codefixes/helpers.ts

```diff
@@ -119,7 +121,7 @@ namespace ts.codefix {
                         signatureDeclarations.push(methodDeclaration);
                     }
                 }
-                else {
+                else if (needsImplementation) {

```

Comment:
Is it possible that `declarations.length > signatures.length` but we don't want to add an implementation?
While I'm here, `TypeChecker.signatureToSignatureDeclaration` has a non-nullable result. Then in `signatureToMethodDeclaration` it's cast to another non-nullable type. Then it's tested for existence!

Reply:
> Is it possible that `declarations.length > signatures.length` but we don't want to add an implementation?

Good catch, but I don't even understand what that check is for and what it's doing. @aozgaa?

Reply:
See this comment (a few lines above): https://github.com/Microsoft/TypeScript/pull/19708/files#diff-6d6034083b71c2ed75493db0fe725ec5R88.

An example (haven't checked by actual debugging) might be

```ts
class C {
     foo(x: number): void;
     foo(x: string): void;
     foo(x: number | string) { console.log(x); }
}

declare class D implements C {
}
```

Then I think adding foo to `D` would be an example.

Reply:
I think this check should be inside `createMethodImplementingSignatures`, which will just omit the body if we are in an ambient context.